addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -347,6 +357,24 @@ export class LinkPopover extends Component {
             return deduceURLfromText(text, this.props.linkElement) || "";
         }
     }
+    getButtonShape() {
+        const shapeToRegex = (shape) => {
+            const parts = shape.trim().split(/\s+/);
+            const regexParts = parts.map((cls) =>
+                ["outline", "fill"].includes(cls) ? `(?=.*\\bbtn-${cls}\\b)` : `(?=.*\\b${cls}\\b)`
+            );
+            return new RegExp(regexParts.join(""));
+        };
+        // Assuming the most matching elements are bottom of the list.
+        for (let i = this.buttonShapeData.length - 1; i >= 0; i--) {
+            const { shape } = this.buttonShapeData[i];
+            const regex = shapeToRegex(shape);
+            if (regex.test(this.props.linkElement.className)) {
+                return shape;
+            }
+        }
```

Comment:
I prefer to use for of
```suggestion
        for (const { shape } of this.buttonShapeData) {
            const regex = shapeToRegex(shape);
            if (regex.test(this.props.linkElement.className)) {
                return shape;
            }
        }
```

Reply:
Why not use the number of part to know which shape chose between "fill outsite" and "fill" ?